### ISSUE TITLE
Conversion of wreath product into matrix group

### ISSUE THREAD

**Issue by joschmitt** [oscar-system/Oscar.jl#3413]
**Is your feature request related to a problem? Please describe.**
Let's take a wreath product of a matrix group with a permutation group:
```
julia> G = matrix_group(QQ[0 1; 1 0])
Matrix group of degree 2
  over rational field

julia> W = wreath_product(G, symmetric_group(2))
<matrix group with 2 generators>
```
Apparently, `W` knows that it is a matrix group on the GAP side. How do I turn it into an OSCAR `MatrixGroup`?

**Describe the solution you'd like**
It would be great, if something like `isomorphism(MatrixGroup, W)` would work.

**Describe alternatives you've considered**
I managed to get what I want using the `Oscar.iso_oscar_gap` magic on the generators of `W`. Would this be the way to go for an `isomorphism` function?

CC @ThomasBreuer @fingolfin

**Comment by thofma** [comment]
Hm, `isomorphism(MatrixGroup, W)` should do something mathematical, that is, finding (implicitly) one faithful matrix representation (over what field exactly?). This might not be the matrix group that you want.

What you want is more a preserving of the matrix group type in the wreath product constructor.

**Comment by joschmitt** [comment]
Just to be clear: I'm not talking about turning any given group into a matrix group. But IF the group is a matrix group on the GAP side (`GAP.Globals.IsMatrixGroup` gives `true`), then I would like to work with it as a matrix group in OSCAR.

If the wreath product method magically returns what I want, I won't complain.

**Comment by ThomasBreuer** [comment]
If the idea is to get the Oscar matrix group that is defined by the matrices in the wreath product then the first attempt to construct this group (with the same underlying GAP object) would be something like `Oscar._oscar_group(W.X, matrix_group(base_ring(W.G), degree(W.H)*degree(W.G)))`.

However, looking at the documentation, there seems to be a subtle question whether `degree(W.H)` or the number of moved points of `W.H` is the right number, depending on how the wreath product was constructed.

**Comment by thofma** [comment]
Yes, I think I understand what you mean. What I mean is that the description of what `isomorphism(MatrixGroup, W)` does should not involve `GAP.Globals.IsMatrixGroup`. It should fit into the following scheme:
```
  isomorphism(::Type{T}, G::GAPGroup) where T <: Union{FPGroup, PcGroup, PermGroup}

  Return an isomorphism from G to a group of type T. An exception is thrown if no such isomorphism exists.
```

**Comment by joschmitt** [comment]
Okay, I'm also happy with a 'type casting' à la `MatrixGroup(W)`.

**Comment by thofma** [comment]
This is the same as `isomorphism(MatrixGroup, W)`, see the docstring lines that I did not paste:
```
  isomorphism(::Type{T}, G::GAPGroup) where T <: Union{FPGroup, PcGroup, PermGroup}

  Return an isomorphism from G to a group of type T. An exception is thrown if no such isomorphism exists.

  Isomorphisms are cached in G, subsequent calls of isomorphism with the same T yield identical results.

  If only the image of such an isomorphism is needed, use T(G).
```

I am not saying that `isomorphism(MatrixGroup, G)` is not allowed to call `IsMatrixGroup` in its implementation. But the expectation should not be, that if one sees `<matrix group with 2 generators>`, then this must be used in `isomorphism(MatrixGroup, G)` or `MatrixGroup(G)`. Or in other words, the "matrix group" in `<matrix group with 2 generators>` is an implementation detail and should not be relied upon IMHO. While we are at it, it would be better if things of type `WreathProductGroup` would be printed differently.

**Comment by fingolfin** [comment]
Yes, printing for these should be adjusted...

which reminds me that my groups printing PR is still open 😱
